packages/driver/src/cross-origin/communicator.ts

```diff
@@ -107,6 +107,33 @@ export class PrimaryOriginCommunicator extends EventEmitter {
       data: preprocessedData,
     }, '*')
   }
+
+  /**
+   * Promisified event sent from the the primary communicator that expects the same event reflected back with the response.
+   * @param {string} event - the name of the event to be sent.
+   * @param {Cypress.ObjectLike} data - any meta data to be sent with the event.
+   * @param options - contains boolean to sync globals
+   * @returns the response from primary of the event with the same name.
+   */
+  toSpecBridgePromise<T> (originPolicy: string, event: string, data?: any) {
+    return new Promise<T>((resolve, reject) => {
+      let timeoutId
+
+      const handler = (result) => {
+        clearTimeout(timeoutId)
+        resolve(result)
+      }
+
+      timeoutId = setTimeout(() => {
+        this.off(event, handler)
+        reject(new Error(`${event} failed to receive a response from ${originPolicy} spec bridge within 1 second.`))
+      }, 1000)
+
+      this.once(event, handler)
```

Comment:
What happens if 2 (or more) of the same event are sent in succession, but resolved in the primary in a different order? Won't the handlers get switched? It might not be a problem with current usage of this, but I worry it's a foot-gun for future uses.

I'd expect there to be a unique ID passed with the message to the primary and it's only resolved here when the same ID is responded to.

Reply:
UUID is a good idea. Let me work on that.